instrumentation-tests/src/androidTest/java/com/mapbox/navigation/instrumentation_tests/core/TripSessionsBillingTest.kt

```diff
@@ -0,0 +1,311 @@
+        val observer = HttpServiceEventsObserver.register()
+
+        mapboxNavigation.setNavigationRoutesAndWaitForUpdate(listOf(route))
+        mapboxNavigation.startTripSessionAndWaitForActiveGuidanceState()
+        mapboxNavigation.stopTripSessionAndWaitForIdleState()
+
+        delay(TimeUnit.SECONDS.toMillis(SESSION_PAUSE_DURATION_SECONDS))
+
+        mapboxNavigation.startTripSessionAndWaitForActiveGuidanceState()
+        mapboxNavigation.clearNavigationRoutesAndWaitForUpdate()
+
+        delay(HTTP_EVENT_APPEARANCE_TIMEOUT_MILLIS)
+
+        val billingRequests = observer.billingRequests()
+        assertEquals(1, billingRequests.size)
+        assertTrue(billingRequests.first().isActiveGuidanceSession)
+
+        assertSessionDuration(0, billingRequests.first().duration)
+    }
+
+    @Test
+    fun testBillingSessionSentForOneSecondLongFreeDrive() = sdkTest {
+        val observer = HttpServiceEventsObserver.register()
+
+        mapboxNavigation.startTripSessionAndWaitForFreeDriveState()
+        delay(TimeUnit.SECONDS.toMillis(SESSION_DURATION_SECONDS))
+        mapboxNavigation.stopTripSessionAndWaitForIdleState()
+        mapboxNavigation.onDestroy()
+
+        delay(HTTP_EVENT_APPEARANCE_TIMEOUT_MILLIS)
+
+        val billingRequests = observer.billingRequests()
+        assertEquals(1, billingRequests.size)
+        assertTrue(billingRequests.first().isFreeDriveSession)
+
+        assertSessionDuration(SESSION_DURATION_SECONDS, billingRequests.first().duration)
+    }
+
+    @Test
+    fun testBillingSessionSentForEmptyFreeDrive() = sdkTest {
+        val observer = HttpServiceEventsObserver.register()
+
+        mapboxNavigation.startTripSessionAndWaitForFreeDriveState()
+        mapboxNavigation.stopTripSessionAndWaitForIdleState()
+        mapboxNavigation.onDestroy()
+
+        delay(HTTP_EVENT_APPEARANCE_TIMEOUT_MILLIS)
+
+        val billingRequests = observer.billingRequests()
+        assertEquals(1, billingRequests.size)
+        assertTrue(billingRequests.first().isFreeDriveSession)
+
+        assertSessionDuration(0, billingRequests.first().duration)
+    }
+
+    @Test
+    fun testOneBillingSessionSentForPausedAndResumedFreeDrive() = sdkTest {
+        val observer = HttpServiceEventsObserver.register()
+
+        mapboxNavigation.startTripSessionAndWaitForFreeDriveState()
+        mapboxNavigation.stopTripSessionAndWaitForIdleState()
+
+        delay(TimeUnit.SECONDS.toMillis(SESSION_PAUSE_DURATION_SECONDS))
+
+        mapboxNavigation.startTripSessionAndWaitForFreeDriveState()
+        mapboxNavigation.stopTripSessionAndWaitForIdleState()
+
+        mapboxNavigation.onDestroy()
+
+        delay(HTTP_EVENT_APPEARANCE_TIMEOUT_MILLIS)
+
+        val billingRequests = observer.billingRequests()
+        assertEquals(1, billingRequests.size)
+        assertTrue(billingRequests.first().isFreeDriveSession)
+
+        assertSessionDuration(0, billingRequests.first().duration)
```

Comment:
Right now you can't distinguish whether you really calculated both parts. Because 0 = 0 + 0 as well as 0 = 0.
I think we should also have a test (instead of or in addition to this one) where both parts are non-zero (like `testOneBillingSessionSentForOneSecondLongPausedAndResumedActiveGuidanceTrip` but for FD).

Reply:
Thank you, good note, I'll fix this test so that session duration is not empty